src/dll/dll.c

```diff
@@ -6,90 +6,131 @@
 #include "cpu.c"
 #include "util.c"
 
+// Define PLOAD as a pointer to a function that takes no arguments and returns a DWORD.
+typedef DWORD(*PLOAD)();
+
+// Handle to the DLL module.
 static HMODULE g_hSelf = NULL;
-static BOOL g_mainThreadFinished = FALSE;
 
-/* Improves accuracy of timeouts and disables power throttling */
-void DisableWindowsThrottling() {
-	/* Will be cleaned up when the process exits */
+// Boolean flag to indicate whether the main thread has finished execution.
+static volatile BOOL g_mainThreadFinished = FALSE; // declared as volatile because it is shared between threads
+
+// Function to disable Windows throttling for better timing accuracy.
+void DisableWindowsThrottling()
+{
+	// This reduces the system timer resolution, improving accuracy of timeouts.
 	timeBeginPeriod(1);
 
+	// Disable power throttling for the process.
 	UtilSetPowerThrottlingState(PROCESS_POWER_THROTTLING_IGNORE_TIMER_RESOLUTION, FALSE);
 	UtilSetPowerThrottlingState(PROCESS_POWER_THROTTLING_EXECUTION_SPEED, FALSE);
 }
 
-/* Hook OsTimeManager::TimeKeeper to apply our changes and prevent them from being overwritten once the game starts */
-DWORD WINAPI VfTimeKeeperThreadProc(LPVOID lpParameter) {
+// Hooked function to apply our changes to the timekeeper.
+DWORD WINAPI VfTimeKeeperThreadProc(LPVOID lpParameter)
+{
 	DisableWindowsThrottling();
 
 	*g_pWowTimerTicksPerSecond = CpuCalibrateTsc();
 	*g_pWowTimerToMilliseconds = 1.0 / (*g_pWowTimerTicksPerSecond * 0.001);
-	/* Align TSC with GetTickCount for CHECK_TIMING_VALUES */
-	*g_pWowTimerOffset = GetTickCount() - (fnWowReadTsc() * *g_pWowTimerToMilliseconds);
+
+	// Align TSC with GetTickCount64 for CHECK_TIMING_VALUES
+	*g_pWowTimerOffset = GetTickCount64() - (fnWowReadTsc() * *g_pWowTimerToMilliseconds);
```

Comment:
This is wrong. VanillaFixes needs to align the timer with `GetTickCount` (even if it has overflowed) because that's what WoW uses by default.